File: 28-implement-strstr/28-implement-strstr.c
int strStr(char * haystack, char * needle)
{
    int len = strlen(haystack),j;
    int l=strlen(needle);
   
    if(l==0)
    return 0;
    
    if(len<l)
    return -1;
    
    for(int i=0;i<len;i++)
    {
        for(j=0;j<l;j++)
        {
            if(haystack[i+j]!=needle[j])
            break;    
        }
        if(j==l)
        return i;
    }  
    return -1;
}

File: 28-implement-strstr/28-implement-strstr.js
var strStr = function (haystack, needle) {
    // Base condition
    if (haystack == null || needle == null) {
        return -1;
    }
    // Special case
    if (haystack === needle) {
        return 0;
    }
    // length of the needle
    const needleLength = needle.length;
    // Loop through the haystack and slide the window
    for (let i = 0; i < haystack.length - needleLength + 1; i++) {
        // Check if the substring equals to the needle
        if (haystack.substring(i, i + needleLength) === needle) {
            return i;
        }
    }
    return -1;
};